libcodechecker/analyze/analysis_manager.py

```diff
@@ -77,6 +80,56 @@ def worker_result_handler(results, metadata, output_path):
 
     metadata['result_source_files'] = source_map
 
+
+def create_dependencies(action):
+    """
+    Transforms the given original build 'command' to a command that, when
+    executed, is able to generate a dependency list.
+    """
+    if 'CC_LOGGER_GCC_LIKE' not in os.environ:
+        os.environ['CC_LOGGER_GCC_LIKE'] = 'gcc:g++:clang:clang++:cc:c++'
+    command = action.original_command.split(' ')
+
+    if any(binary_substring in command[0] for binary_substring
+           in os.environ['CC_LOGGER_GCC_LIKE'].split(':')):
+        # gcc and clang can generate makefile-style dependency list.
+        command = [command[0], '-E', '-M', '-MQ', '__dummy'] + command[1:]
+
+        try:
```

Comment:
I think to check that something is in list is better than catching the exceptions in this case:

```python
if '-o' in command:
    option_index = command.index('-o')
elif '--output' in command:
    option_index = command.index('--output')
else:
    option_index = None
```